web3/utils/math.py

```diff
@@ -0,0 +1,12 @@
+import math
+
+
+def percentile(values, percentile):
+    if not values:
+        return None
```

Comment:
This looks like a place where an exception should be thrown.  Return value checking is inherently error prone.